diff -urNp -x '*.orig' guile-1.8.8.org/libguile/guile-func-name-check.in guile-1.8.8/libguile/guile-func-name-check.in
--- guile-1.8.8.org/libguile/guile-func-name-check.in	2010-12-13 18:24:40.000000000 +0100
+++ guile-1.8.8/libguile/guile-func-name-check.in	2023-02-02 08:57:48.934211013 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/awk -f
+#!/bin/awk -f
 #
 #  Copyright (C) 2000, 2001, 2006 Free Software Foundation, Inc.
 # 
