--- guile-1.4/libguile/Makefile.am~	Mon Feb 25 06:30:37 2002
+++ guile-1.4/libguile/Makefile.am	Mon Feb 25 06:30:37 2002
@@ -93,11 +93,6 @@
     filesys.c posix.c net_db.c socket.c \
     ramap.c unif.c debug-malloc.c
 
-## In next release, threads will be factored out of libguile.
-## Until then, the machine specific headers is a temporary kludge.
-OMIT_DEPENDENCIES = libguile.h ltdl.h \
-    axp.h hppa.h i386.h ksr.h m88k.h mips.h sparc.h vax.h
-
 ## This is kind of nasty... there are ".c" files that we don't want to
 ## compile, since they are #included in threads.c.  So instead we list
 ## them here.  Perhaps we can deal with them normally once the merge
--- guile-1.4/qt/Makefile.am~	Mon Feb 25 06:31:30 2002
+++ guile-1.4/qt/Makefile.am	Mon Feb 25 06:31:30 2002
@@ -36,8 +36,6 @@
 libqthreads_la_DEPENDENCIES = qtmds.lo qtmdc.lo
 libqthreads_la_LDFLAGS = -rpath $(libdir)
 
-OMIT_DEPENDENCIES = axp.h hppa.h i386.h ksr.h m88k.h mips.h sparc.h vax.h
-
 .s.lo:
 	$(LIBTOOL) --mode=compile $(COMPILE) $(qthread_asflags) -c $<
 qtmds.s:
