--- guile-1.6.4/libguile/gc.c.orig	2004-05-24 21:54:13.755811688 +0200
+++ guile-1.6.4/libguile/gc.c	2004-05-24 21:53:44.038032247 +0200
@@ -1129,6 +1129,9 @@
   /* FIXME: we should have a means to register C functions to be run
    * in different phases of GC
    */
+  /* don't try to use scm_subr_table during its reallocation
+   * (when scm_subr_table still points to old memory area after realloc() call) */
+  if (strcmp(what, "scm_subr_table") != 0)
   scm_mark_subr_table ();
 
 #ifndef USE_THREADS
