--- guile-1.6.0/libguile/gc_os_dep.c.sopwith	2003-02-11 08:53:15.000000000 -0500
+++ guile-1.6.0/libguile/gc_os_dep.c	2003-02-11 09:01:43.000000000 -0500
@@ -203,6 +203,10 @@
 #    define POWERPC
 #    define mach_type_known
 # endif
+# if defined(LINUX) && defined(__powerpc64__)
+#    define POWERPC64
+#    define mach_type_known
+# endif
 # if defined(LINUX) && defined(__mc68000__)
 #    define M68K
 #    define mach_type_known
@@ -591,6 +595,19 @@
 #   endif
 # endif
 
+# ifdef POWERPC64
+#   define MACH_TYPE "POWERPC64"
+#   define ALIGNMENT 8
+#   define CPP_WORDSZ 64
+#   define HEURISTIC1
+#   define DYNAMIC_LOADING
+#   undef STACK_GRAN
+#   define STACK_GRAN 0x10000000
+#   define LINUX_DATA_START
+    extern int _end;
+#   define DATAEND (&_end)
+# endif
+
 # ifdef POWERPC
 #   define MACH_TYPE "POWERPC"
 #   ifdef MACOS
