diff -ru guile-1.4.old/libguile/net_db.c guile-1.4/libguile/net_db.c
--- guile-1.4.old/libguile/net_db.c	Wed Nov 15 12:33:51 2000
+++ guile-1.4/libguile/net_db.c	Wed Nov 15 12:34:45 2000
@@ -80,9 +80,8 @@
 
 #ifndef STDC_HEADERS
 int close ();
-#endif /* STDC_HEADERS */
-
 extern int inet_aton ();
+#endif /* STDC_HEADERS */
 
 SCM_DEFINE (scm_inet_aton, "inet-aton", 1, 0, 0, 
             (SCM address),
Only in guile-1.4/libguile: net_db.c~
