--- guile-1.6.5/libguile/gc_os_dep.c.orig	2004-06-16 00:55:31.000000000 +0200
+++ guile-1.6.5/libguile/gc_os_dep.c	2004-11-01 23:45:31.105332992 +0100
@@ -227,6 +227,10 @@
 #    define POWERPC
 #    define mach_type_known
 # endif
+# if defined(LINUX) && defined(__powerpc64__)
+#    define POWERPC64
+#    define mach_type_known
+# endif
 # if defined(LINUX) && defined(__mc68000__)
 #    define M68K
 #    define mach_type_known
@@ -371,6 +375,10 @@
 #    define S370
 #    define mach_type_known
 # endif
+# if defined(__x86_64__) && defined(LINUX)
+#    define AMD64
+#    define mach_type_known
+# endif
 # if defined(__GNU__)
 #    define I386
 #    define GNU
@@ -631,6 +639,19 @@
 #   endif
 # endif
 
+# ifdef POWERPC64
+#   define MACH_TYPE "POWERPC64"
+#   define ALIGNMENT 8
+#   define CPP_WORDSZ 64
+#   define HEURISTIC1
+#   define DYNAMIC_LOADING
+#   undef STACK_GRAN
+#   define STACK_GRAN 0x10000000
+#   define LINUX_DATA_START
+    extern int _end;
+#   define DATAEND (&_end)
+# endif
+
 # ifdef POWERPC
 #   define MACH_TYPE "POWERPC"
 #   ifdef MACOS
@@ -1146,6 +1167,18 @@
 #   endif
 # endif
 
+# ifdef AMD64
+#   define MACH_TYPE "AMD64"
+#   define OS_TYPE "LINUX"
+#   define ALIGNMENT 8
+#   define CPP_WORDSZ 64
+#   define DYNAMIC_LOADING
+#   define LINUX_DATA_START
+extern int _end;
+#   define DATAEND (&_end)
+#   define STACKBOTTOM ((ptr_t) 0x7fc0000000l)
+# endif
+
 # ifdef IA64
 #   define MACH_TYPE "IA64"
 #   define ALIGN_DOUBLE
