--- guile-1.6.4/libguile/gc.c.orig	2003-04-20 20:16:59.000000000 +0200
+++ guile-1.6.4/libguile/gc.c	2004-05-31 23:21:10.807334759 +0200
@@ -2048,9 +2048,14 @@
     scm_must_free() won't take NULL.
   */
   scm_mallocated += size - old_size;
-  SCM_SYSCALL (ptr = realloc (where, size));
 
+  /*
+    The realloc must be after check_mtrigger(), since realloc might
+    munge the old block of memory, which will be scanned by GC.
+   */
   check_mtrigger (what);
+
+  SCM_SYSCALL (ptr = realloc (where, size));
   
   if (NULL != ptr)
     {
