--- guile-1.6.4/libguile/gc_os_dep.c.orig	2004-05-31 23:22:31.000000000 +0200
+++ guile-1.6.4/libguile/gc_os_dep.c	2004-06-01 00:18:13.331037133 +0200
@@ -1158,7 +1158,11 @@
 #       define CPP_WORDSZ 64
 	/* This should really be done through /proc, but that	*/
 	/* requires we run on an IA64 kernel.			*/
+#ifdef __x86_64__
+#	define STACKBOTTOM ((ptr_t) 0x7fc0000000l)
+#else /* really IA64 */
 #       define STACKBOTTOM ((ptr_t) 0xa000000000000000l)
+#endif
 	/* We also need the base address of the register stack	*/
 	/* backing store.  There is probably a better way to	*/
 	/* get that, too ...					*/
@@ -1865,8 +1869,11 @@
 
 #endif /* LINUX_STACKBOTTOM */
 
+extern void *__libc_stack_end;
+
 void *scm_get_stack_base()
 {
+    return __libc_stack_end;
     word dummy;
     void *result = &dummy;  /* initialize to silence compiler */
 
