--- guile-1.8.1/test-suite/tests/popen.test.orig	2006-08-27 23:05:26.000000000 +0200
+++ guile-1.8.1/test-suite/tests/popen.test	2006-10-10 01:12:27.114961750 +0200
@@ -82,7 +82,7 @@
 	   (port (with-error-to-port (cdr pair)
 		   (lambda ()
 		     (open-input-pipe
-		      "exec 1>/dev/null; echo closed 1>&2; exec 2>/dev/null; sleep 999")))))
+		      "exec 1>/dev/null; echo closed 1>&2; exec 2>/dev/null; sleep 99")))))
       (close-port (cdr pair))   ;; write side
       (and (char? (read-char (car pair))) ;; wait for child to do its thing
 	   (char-ready? port)
@@ -132,7 +132,7 @@
 	      (port (with-error-to-port (cdr pair)
 		      (lambda ()
 			(open-output-pipe
-			 "exec 0</dev/null; echo closed 1>&2; exec 2>/dev/null; sleep 999")))))
+			 "exec 0</dev/null; echo closed 1>&2; exec 2>/dev/null; sleep 99")))))
 	 (close-port (cdr pair))   ;; write side
 	 (and (char? (read-char (car pair))) ;; wait for child to do its thing
 	      (catch 'system-error
